packages/react-i18n/src/decorator.tsx

```diff
@@ -34,6 +34,11 @@ type TranslationMap = {
   [key: string]: MaybePromise<TranslationDictionary | undefined>;
 };
 
+type StaticFields<Object> = {
+  [Key in keyof Object]: Key extends 'prototype' ? never : Key
+}[keyof Object];
+export type Statics<Object> = Pick<Object, StaticFields<Object>>;
```

Comment:
Shoot sucks that we have this in multiple places, we should maybe put these types in `@shopify/useful-types` instead.